sklearn/preprocessing/_polynomial.py

```diff
@@ -962,6 +963,8 @@ def transform(self, X):
         n_samples, n_features = X.shape
         n_splines = self.bsplines_[0].c.shape[1]
         degree = self.degree
+        mean_X = np.mean(X, axis=0, dtype=np.float64)
+        var_X = np.var(X, axis=0, dtype=np.float64)
```

Comment:
We can compute both at the same time using `_incremental_mean_and_var`. This is a bit low level and requires passing zeros arrays for previous mean and variance along with zeros for `last_sample_count`.

Also, I would rather user the input dtype for the computation and hope that `_incremental_mean_and_var` is numerically stable enough.